aplus/celery.py

```diff
@@ -29,3 +30,5 @@ def enroll():
     )
     for i in courses:
         i.enroll_from_sis()
+        if hasattr(settings, 'SIS_ENROLL_DELAY'):
```

Comment:
I have preferred the style that all settings have a default value in settings.py, for example, `None`. Then, you wouldn't use `if hasattr(...)` but `if settings.SIS_ENROLL_DELAY:`. Any reason why you like `hasattr`?

At least, default values have the advantage that you can see the possible settings in settings.py. With your solution, settings.py has no sign of `SIS_ENROLL_DELAY` when someone wants to list all the settings that should be maybe set in an installation. The documentation and examples lessen that downside. The idea that some admin would need to read the code in order to even find the possible settings is horrific if the settings weren't listed anywhere easily.

Reply:
There is also the perspective of a developer. Is another developer going to search for possible settings in the examples or do they assume that all settings are visible in settings.py because they thought it is the best practice?